designs/2022-unused-disable-directive-flexible-config/README.md

```diff
@@ -0,0 +1,228 @@
+
+We will also change the default behavior to warn on unused disable directives so that all users (unless they opt-out) will benefit from the detection (and automatic removal with `--fix`) of unused disable directives.
+
+## Detailed Design
+
+<!--
+   This is the bulk of the RFC.
+
+   Explain the design with enough detail that someone familiar with ESLint
+   can implement it by reading this document. Please get into specifics
+   of your approach, corner cases, and examples of how the change will be
+   used. Be sure to define any new terms in this section.
+-->
+
+How the new "explicit severity" design works:
+
+- The config and CLI options are both changed to also accept standard severity level string values: `off`, `warn`, `error` (or a the corresponding number for each level)
+- If both options are present, the CLI option takes precedence over (overrides) the config option
+- If only one of the options is present, the value of the present option is used
+- If neither option is present, the default value (`warn`) is used
+- Note: There's only one underlying setting, and the config and CLI options are just two different ways of controlling it
+
+Boolean values will continue to be allowed with no change in behavior, as summarized below.
+
+Allowed values for the config file option:
+
+- `reportUnusedDisableDirectives: true` - same behavior as today (`warn`)
+- `reportUnusedDisableDirectives: false` - same behavior as today (`off`)
+- `reportUnusedDisableDirectives: "error"` - new value
+- `reportUnusedDisableDirectives: "warn"` - new value
+- `reportUnusedDisableDirectives: "off"` - new value
+
+Allowed values for the CLI option:
+
+- `--report-unused-disable-directives` - same behavior as today (`error`) (common CLI shorthand to omit the value)
+- `--report-unused-disable-directives=true` - same behavior as today (`error`)
+- `--report-unused-disable-directives=false` - same behavior as today (`off`)
+- `--report-unused-disable-directives=error` - new value
+- `--report-unused-disable-directives=warn` - new value
+- `--report-unused-disable-directives=off` - new value
+
+### Phases
+
+The implementation of this RFC will likely involve two phases:
+
+1. Phase 1: A non-breaking change to add support for the new severity levels.
+2. Phase 2: A breaking change intended for a major release in which the default behavior is changed to `warn`.
+
+This allows us to get the new functionality out to users as soon as possible as a smaller change with reduced risk, while also keeping the breaking change small and focused.
+
+### Code Changes
+
+- `conf/flat-config-schema.js` - also support string / number type
+- `conf/default-cli-options.js` - use new default
+- `lib/options.js` - also support string / number type and use new default for the option
+  - Something like `type: "Boolean|String|Number"` and `enum: ["true", "false", "error", "warn", "off", "0", "1", "2"]`
```

Comment:
When set, CLI/API option overrides the config option. I think we should keep the same CLI/API default (`undefined`/`null`) and rather define the new default in the config schema.

Reply:
I updated to mention that `lib/config/flat-config-schema.js` is where the new default would be applied. Is that what you meant?

Reply:
> lib/config/flat-config-schema.js

Yes, but I think the current version of this RFC says that the default will be applied in `lib/linter/linter.js`?

Reply:
Okay, I updated to mention the default would be applied in `lib/config/flat-config-schema.js`. I don't see any defaults in that file yet so perhaps you can comment here or in the implementation https://github.com/eslint/eslint/pull/17212 to let me know how to set a default there.

Reply:
Defaults are here:

https://github.com/eslint/eslint/blob/main/lib/config/default-config.js

Reply:
Okay, I added:

```md
- `lib/config/default-config.js` - add new default value of `warn` in phase 2
```

Does the rest of the RFC look good in terms of getting to final commenting period?

Reply:
Looks good to me, but perhaps we could treat phase 2 as non-breaking. I'll leave a question in the Phases section.